### PR TITLE
adding transparency to window builder

### LINKED ISSUE #3032: Allow window to be transparent
## What problem does this solve or what need does it fill?

For an example this feature would help users build desktop widgets or something similar to this game: https://samperson.itch.io/desktop-goose

## What solution would you like?

A way to set the transparency of the window on application init or in a system

## What alternative(s) have you considered?

I've tried accessing the `ResMut<Windows>` but no api is exposed to modify the winit window transparency 

## Additional context

It seems that winit allows the transparency to be set via the WindowBuilder but not after the creation of the window.
winit docs:
https://docs.rs/winit/0.25.0/winit/window/struct.WindowBuilder.html#method.with_transparent
https://docs.rs/winit/0.25.0/winit/window/struct.WindowAttributes.html#structfield.transparent

### PR DESCRIPTION
**Pull request by louisgjohnson**
# Objective

Fixes #3032

Allowing a user to create a transparent window 

## Solution

I've allowed the transparent bool to be passed to the winit window builder

### CODE REVIEW

File: examples/window/transparent_window.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/// - iOS / Android / Web: Unsupported.
+/// - OSX: Not working as expected.
+use bevy::{prelude::*, render::pass::ClearColor, window::WindowDescriptor};
+
+fn main() {
+    App::new()
+        // rgba value needs to be [0, 0, 0, 0], otherwise some color will bleed through
+        .insert_resource(ClearColor(Color::NONE))
+        .insert_resource(WindowDescriptor {
+            // setting transparent allows the window to become transparent when clear color has the correct value
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply examples/window/transparent_window.rs:11]: Yeah, that's just a typo.

**Comment by louisgjohnson** [comment]
> I left some nits on the commenting, but I'm now happy with the quantity and quality (other than the nits) of the documentation for this PR :)

Whoops, fixed up the typo's

**Review by IceSentry — APPROVED** [review]

File: crates/bevy_window/src/window.rs [hunk 2]
```diff
@@ -534,6 +534,7 @@ pub struct WindowDescriptor {
     pub cursor_visible: bool,
     pub cursor_locked: bool,
     pub mode: WindowMode,
+    pub transparent: bool,
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:537]: I would like to document the cross-platform limitations here as well, to improve visibility.

IMO this should be the canonical place for it,  and the example should have a brief link to these docs.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors try

**Comment by inodentry** [comment]
Why was this PR closed?

**Comment by alice-i-cecile** [comment]
@inodentry, this was closed in favor of #3105, which is just a branch-fixed version of this PR.

**Pull request closed without merging**